docs/sources/next/using-k6/javascript-compatibility-mode.md

```diff
@@ -1,44 +1,71 @@
 ---
-title: JavaScript Compatibility Mode
+title: JavaScript compatibility mode
+menuTitle: JavaScript mode
 excerpt: 'k6 supports running test scripts with different ECMAScript compatibility modes using --compatibility-mode'
 _build:
   list: false
 weight: 19
 ---
 
-# JavaScript Compatibility Mode
+# JavaScript compatibility mode
 
-You can run test scripts with different ECMAScript compatibility modes with the
-`run --compatibility-mode` CLI option or `K6_COMPATIBILITY_MODE` environment variable.
+You can write k6 scripts in various ECMAScript versions:
 
-Currently two modes are available:
+- ES6+ JavaScript with ES modules (ESM).
+- Plain old JavaScript (ES5.1+) with CommonJS modules.
 
-## Base
+k6 supports both module types and most ES6+ features in all k6 execution modes: local, distributed, and cloud.
+
+To enable ES module support, k6 uses [Babel](https://babeljs.io/) internally to transform ESM to CommonJS. The process is as follows:
+
+![Babel transformation in k6](/media/docs/k6-oss/diagram-grafana-k6-babel-pipeline.png)
+
+When executing large scripts, the k6 Babel transformation can slightly delay the startup time. Because of that, some users prefer to bundle their test code outside k6. For this reason, k6 offers two JavaScript compatibility modes:
+
+- [Extended mode](#extended-mode): The default option, supporting ESM and most ES6+ features.
+- [Base mode](#base-mode): Limited to plain old JavaScript (ES5.1) and CommonJS, excluding the Babel step.
```

Comment:
I guess here it is important to note that:
1. Extented mode at this poitn only adds ESM support and sets `global` (node's global variable) to have the same value as `globalThis` (which is the specificaiton global variables)
2. base in practice support way more than ES5.1. Pretty sure anything that isn't ESM from ES6 is supported + class stuff from as late as ES2022. And also a ton of other small and not so small features.

Not certain if we want to try to convey this though as ... hopefully soon ESM will be supported without babel and we will have this working through out.

Reply:
Fixed by https://github.com/grafana/k6-docs/pull/1489/commits/8a367ab803b7364a29eba76e813c9fea17c79a4a

> soon ESM will be supported without babel

Then, we will update or remove this article.